### PR TITLE
[WIP] refactor(legacy support): use unified component

### LINKED ISSUE #71: Cleanup/Refactoring
Before I start working on a PR I wanted to discuss this here with you folks. I played a bit recently with this `-in-element` thing, and did this one: https://github.com/kaliber5/ember-in-element-polyfill. This together with https://github.com/DockYard/ember-maybe-in-element would allow to simplify the code base quite a bit. 

Especially I *think* it would allow to drop all the special legacy handling here (effectively moving it to upstream dependencies), so we have just a single unified component that works with Ember 1.13+. The missing hash helper could be polyfilled with https://github.com/cibernox/ember-hash-helper-polyfill btw.

This however does not work all the way down to Ember 1.11. So before hitting stable, I wonder if the minimum Ember version could be raised to at least 1.13 (which is already quite antique! 😬). Some 0.x version could still be supported for Ember 1.11 (probably bug fixes only).

What do you think?

### PR DESCRIPTION
**Pull request by simonihmig**
* uses `in-element-polyfill` and `maybe-in-element` AST transforms to allow to use `in-element` with legacy support back to Ember 1.13 (ember-wormhole based)
* Cleaned up unnecessary legacy components

Closes #71

BREAKING CHANGE:
Support for Ember <1.13 has been dropped!

### CODE REVIEW

File: tests/integration/components/ember-popper/action-test.js [hunk 1]
```diff
@@ -2,51 +2,48 @@ import { moduleForComponent, test } from 'ember-qunit';
...
-      .then(wait)
-      .then(() => {
-        assert.equal(called, 1, 'onUpdate action has been called after event');
-        return wait();
-      });
-  });
-}
+  return wait()
+    .then(() => triggerEvent(document.querySelector('body'), 'scroll'))
+    .then(() => wait())
```
Comment (simonihmig) [inline tests/integration/components/ember-popper/action-test.js]: FYI: this change (`wait` -> `() => wait()`) is unrelated to my changes but was required, as the latest `ember-test-helpers` was causing this error when some argument (here the resolved promise) is passed to `wait`: https://travis-ci.org/kybishop/ember-popper/jobs/328856339#L547.

See https://github.com/emberjs/ember-test-helpers/blob/9eb3e32f8340d01704e950f317b282f07073f1e9/addon-test-support/%40ember/test-helpers/settled.js#L169-L174

Don't know if this should be regarded a regression of `ember-test-helpers/wait` or not? 🤔  /cc @rwjblue

**Comment by simonihmig** [comment]
@pzuraq any idea about this error? https://travis-ci.org/kybishop/ember-popper/jobs/328857614#L542

Happens only for ember-try scenarios with Ember <= 2.12 🤔

File: tests/integration/components/ember-popper/action-test.js (hunk 1 above)
Comment (rwjblue) [reply tests/integration/components/ember-popper/action-test.js]: Ya, can you report over there so we can dig into it?

Comment (simonihmig) [reply tests/integration/components/ember-popper/action-test.js]: https://github.com/emberjs/ember-test-helpers/issues/300

**Comment by pzuraq** [comment]
Something is going wrong with the legacy class transform. I'll see if I can debug sometime soon, swamped at the moment

**Comment by simonihmig** [comment]
Closing this as master has diverged too much. Another PR will follow...

**Pull request closed without merging**